src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -247,7 +249,7 @@ && getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROL
                 useJDKSerialization
             );
 
-            if (isActionTraceEnabled()) {
+            if (isActionTraceEnabled() || actionTraceForTesting) {
```

Comment:
Lets change how `isActionTraceEnabled` is declared so we can inject it - so its controllable in test code.  I'd recommend using a 'Supplier<Boolean>' in class construction.  This allows the removal of `actionTraceForTesting`, what do you think?

Reply:
Sure I will swap it over.